pepdbagent/modules/sample.py

```diff
@@ -0,0 +1,320 @@
+)
+from pepdbagent.exceptions import SampleNotFoundError
+
+from pepdbagent.db_utils import BaseEngine, Samples, Projects
+
+_LOGGER = logging.getLogger(PKG_NAME)
+
+
+class PEPDatabaseSample:
+    """
+    Class that represents Project in Database.
+
+    While using this class, user can create, retrieve, delete, and update projects from database
+    """
+
+    def __init__(self, pep_db_engine: BaseEngine):
+        """
+        :param pep_db_engine: pepdbengine object with sa engine
+        """
+        self._sa_engine = pep_db_engine.engine
+        self._pep_db_engine = pep_db_engine
+
+    def get(
+        self,
+        namespace: str,
+        name: str,
+        sample_name: str,
+        tag: str = DEFAULT_TAG,
+        raw: bool = False,
+    ) -> Union[peppy.Sample, dict, None]:
+        """
+        Retrieve sample from the database using namespace, name, tag, and sample_name
+
+        :param namespace: namespace of the project
+        :param name: name of the project (Default: name is taken from the project object)
+        :param tag: tag (or version) of the project.
+        :param sample_name: sample_name of the sample
+        :param raw: return raw dict or peppy.Sample object
+        :return: peppy.Project object with found project or dict with unprocessed
+            PEP elements: {
+                name: str
+                description: str
+                _config: dict
+                _sample_dict: dict
+                _subsample_dict: dict
+            }
+        """
+        statement_sample = select(Samples).where(
+            and_(
+                Samples.project_id
+                == select(Projects.id)
+                .where(
+                    and_(
+                        Projects.namespace == namespace,
+                        Projects.name == name,
+                        Projects.tag == tag,
+                    ),
+                )
+                .scalar_subquery(),
+                Samples.sample_name == sample_name,
+            )
+        )
+        project_config_statement = select(Projects.config).where(
+            and_(
+                Projects.namespace == namespace,
+                Projects.name == name,
+                Projects.tag == tag,
+            )
+        )
+
+        with Session(self._sa_engine) as session:
+            result = session.scalar(statement_sample)
+            if result:
+                if not raw:
+                    config = session.execute(project_config_statement).one_or_none()[0]
+                    project = peppy.Project().from_dict(
+                        pep_dictionary={
+                            "name": name,
+                            "description": config.get("description"),
+                            "_config": config,
+                            "_sample_dict": [result.sample],
+                            "_subsample_dict": None,
+                        }
+                    )
+                    return project.samples[0]
```

Comment:
I believe that it works, but curious why we get the first one here?

Reply:
because this project will always have only one sample